### ISSUE TITLE
Connect and Web UI submit access requests with different durations

### ISSUE THREAD

**Issue by ravicious** [gravitational/teleport#42562]
Steps to reproduce:

1. Set up a user account for resource access requests as described in the web test plan.
2. Make the same access request from Connect and Web UI.

The duration of the request coming from Connect will be almost 12h, while the duration of the request from the Web UI will be 4 minutes.

https://github.com/gravitational/teleport/assets/27113/96ae99d2-d21c-4257-a552-6b0999988a18

**Comment by kimlisa** [comment]
oh that's so weird... i run into similar problem in the web UI where

- requesting from the access request page gives me the correct max (12 hours, fyi this is your session TTL, and it will be lower the longer you've been logged in)
- requesting from the resource list gives me 8 hours

**Comment by gzdunek** [comment]
I think I know what's going on.
We used to make the access request dry run call when the checkout sidebar opened. However, in the unified resources it is opened permanently, so we make the call only when the component mounts (so we send an empty request and it returns that 4 minutes duration). 

To fix this, we should make the dry run call every time the user adds a resource. 
It also looks like we had a bug even previously, because we didn't make the call if the resource was removed from a request. I don't think this is correct, and we should always call dry run when the number of added resources changes, correct? cc @kimlisa
